route-rs-runtime/src/link/mod.rs

```diff
@@ -33,7 +33,14 @@ mod blackhole_link;
 pub use self::blackhole_link::*;
 
 /// The type that all links take as input, or provide as output. This allows links to be combinatorial
-pub type PacketStream<Input> = Box<dyn Stream<Item = Input, Error = ()> + Send>;
+pub type PacketStream<Input> = Box<dyn Stream<Item = Input, Error = ()> + 'static + Send>;
+pub type TokioRunnable = Box<dyn Future<Item = (), Error = ()> + 'static + Send>;
```

Comment:
I'd like to not have the `Future` be boxed, but was running into the `trait objects without an explicit 'dyn' are deprecated` stuff.

Reply:
Yeah, I figured this would happen.  Good news though is that the function call stack for it will look something like 
poll()..poll()..poll()..poll(). Before, the last 3 polls where already dyn since they were streams.  Now they are all dyn. Doesn't seem like a huge cost to me for the flexibility we gain. I assume the CPU hardware will do a good job prefetching the fn instructions, especially when you do it quickly on repeat over and over.

Reply:
Ideally we wouldn't need Dyn's a guess? One thing to consider though is that if we do manage to monomorph our functions, the size of the router may go up quite a bit.

Reply:
Resolving this as fine 🔥 👌